streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/LegacySubscriptionInfoSerde.java

```diff
@@ -0,0 +1,276 @@
+                                       final UUID processId,
+                                       final Set<TaskId> prevTasks,
+                                       final Set<TaskId> standbyTasks,
+                                       final String userEndPoint) {
+        if (latestSupportedVersion == UNKNOWN && (version < 1 || version > 2)) {
+            throw new IllegalArgumentException(
+                "Only versions 1 and 2 are expected to use an UNKNOWN (-1) latest supported version. " +
+                    "Got " + version + "."
+            );
+        } else if (latestSupportedVersion != UNKNOWN && (version < 1 || version > latestSupportedVersion)) {
+            throw new IllegalArgumentException(
+                "version must be between 1 and " + latestSupportedVersion + "; was: " + version
+            );
+        }
+        usedVersion = version;
+        this.latestSupportedVersion = latestSupportedVersion;
+        this.processId = processId;
+        this.prevTasks = prevTasks;
+        this.standbyTasks = standbyTasks;
+        // Coerce empty string to null. This was the effect of the serialization logic, anyway.
+        this.userEndPoint = userEndPoint == null || userEndPoint.isEmpty() ? null : userEndPoint;
+    }
+
+    public int version() {
+        return usedVersion;
+    }
+
+    public int latestSupportedVersion() {
+        return latestSupportedVersion;
+    }
+
+    public UUID processId() {
+        return processId;
+    }
+
+    public Set<TaskId> prevTasks() {
+        return prevTasks;
+    }
+
+    public Set<TaskId> standbyTasks() {
+        return standbyTasks;
+    }
+
+    public String userEndPoint() {
+        return userEndPoint;
+    }
+
+    /**
+     * @throws TaskAssignmentException if method fails to encode the data
+     */
+    public ByteBuffer encode() {
+        if (usedVersion == 3 || usedVersion == 4 || usedVersion == 5) {
+            final byte[] endPointBytes = prepareUserEndPoint(this.userEndPoint);
+
+            final ByteBuffer buf = ByteBuffer.allocate(
+                4 + // used version
+                    4 + // latest supported version version
+                    16 + // client ID
+                    4 + prevTasks.size() * 8 + // length + prev tasks
+                    4 + standbyTasks.size() * 8 + // length + standby tasks
+                    4 + endPointBytes.length
+            );
+
+            buf.putInt(usedVersion); // used version
+            buf.putInt(LATEST_SUPPORTED_VERSION); // supported version
+            encodeClientUUID(buf, processId());
+            encodeTasks(buf, prevTasks);
+            encodeTasks(buf, standbyTasks);
+            encodeUserEndPoint(buf, endPointBytes);
+
+            buf.rewind();
+
+            return buf;
+        } else if (usedVersion == 2) {
+            final byte[] endPointBytes = prepareUserEndPoint(this.userEndPoint);
+
+            final ByteBuffer buf = ByteBuffer.allocate(
+                4 + // version
```

Comment:
nit: format should align with L125

Reply:
This is the way IDEA formats multiline mathematical expressions.